### PR TITLE
cop: RPN function IN()

### PR DESCRIPTION
**Pull request by breeswish**
## What have you changed? (mandatory)

This PR implements RPN function `IN()`.

## What are the type of the changes? (mandatory)

- Improvement (change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

New unit tests.

### CODE REVIEW

File: src/coprocessor/dag/rpn_expr/types/test_util.rs [hunk 1]
```diff
@@ -36,6 +36,17 @@ impl RpnFnScalarEvaluator {
         self
     }
 
+    pub fn push_params<T, S>(mut self, values: S) -> Self
+    where
+        T: Into<ScalarValue>,
+        S: std::iter::IntoIterator<Item = T>,
```
Comment (sticnarf) [inline src/coprocessor/dag/rpn_expr/types/test_util.rs]: `IntoIterator` is in Rust's prelude.

And, what about `values: impl IntoIterator<Item = impl Into<ScalarValue>>`?

File: src/coprocessor/dag/rpn_expr/impl_compare.rs [hunk 2]
```diff
@@ -207,6 +207,34 @@ pub fn coalesce<T: Evaluable>(args: &[&Option<T>]) -> Result<Option<T>> {
     Ok(None)
 }
 
+#[rpn_fn(varg)]
+#[inline]
+pub fn compare_in<T: Evaluable + Eq>(args: &[&Option<T>]) -> Result<Option<Int>> {
```
Comment (sticnarf) [inline src/coprocessor/dag/rpn_expr/impl_compare.rs]: Can we add `Eq` to the trait bound of `Evaluable`?

Comment (breeswish) [reply src/coprocessor/dag/rpn_expr/impl_compare.rs]: Too many other bounds to add if we add one... For example, there are also PartialEq, PartialOrd, etc..

File: src/coprocessor/dag/rpn_expr/types/test_util.rs (hunk 1 above)
Comment (breeswish) [reply src/coprocessor/dag/rpn_expr/types/test_util.rs]: Wow. I don't know this syntax is valid. I will give a try.

Comment (breeswish) [reply src/coprocessor/dag/rpn_expr/types/test_util.rs]: Interesting..For some reasons I don't know, this leads to:

```
error[E0282]: type annotations needed
   --> src/coprocessor/dag/rpn_expr/impl_compare.rs:659:18
    |
659 |                 .push_params(args)
    |                  ^^^^^^^^^^^ cannot infer type for `T`

error: aborting due to previous error
```

Comment (sticnarf) [reply src/coprocessor/dag/rpn_expr/types/test_util.rs]: No problem here. I'm curious about why there is type `T` when the function looks like ` pub fn push_params(mut self, values: impl IntoIterator<Item = impl Into<ScalarValue>>) -> Self `

**Comment by breeswish** [comment]
@lonng @H-ZeX PTAL, thanks

**Pull request merged**